en/ros/external_position_estimation.md

```diff
@@ -1,114 +1,220 @@
+Both VIO and MoCap determine a vehicle's *pose* (position and attitude) from "visual" information.
+The main difference between them is the frame perspective: 
+- VIO uses *onboard sensors* to get pose data from the vehicle's perspective (see [egomotion](https://en.wikipedia.org/wiki/Visual_odometry#Egomotion)).
+- MoCap uses a system of *off-board cameras* to get vehicle pose data in a 3D space (i.e. it is an external system that tells the vehicle its pose).
 
-Position estimates can be sent both from an onboard computer as well as from offboard (example : VICON).  This data is used to update its local position estimate relative to the local origin. Heading from the vision/motion capture system can also be optionally integrated by the attitude estimator.
+Pose data from either type of system can be used to update a PX4-based autopilot's local position estimate (relative to the local origin) and also can optionally also be fused into the vehicle attitude estimation.
 
-The system can then be used for applications such as position hold indoors or waypoint navigation based on vision.
+This topic explains how to configure a PX4-based system to get data from MoCap/VIO systems (either via ROS or some other MAVLink system) and more specifically how to set up MoCap systems like VICON and Optitrack, and vision-based estimation systems like [ROVIO](https://github.com/ethz-asl/rovio), [SVO](https://github.com/uzh-rpg/rpg_svo) and [PTAM](https://github.com/ethz-asl/ethzasl_ptam)).
 
-For vision, the MAVLink message used to send the pose data is [VISION_POSITION_ESTIMATE](https://mavlink.io/en/messages/common.html#VISION_POSITION_ESTIMATE) and the message for all motion capture systems is [ATT_POS_MOCAP](https://mavlink.io/en/messages/common.html#ATT_POS_MOCAP) messages.
+> **Note** The instructions differ depending on whether you are using the EKF2 or LPE estimator.
 
-The mavros ROS-MAVLink interface has default implementations to send these messages. They can also be sent using pure C/C++ code and direct use of the MAVLink library. The ROS topics are: `/mavros/mocap/pose` for mocap systems and `/mavros/vision_pose/pose` for vision. Check [mavros_extras](http://wiki.ros.org/mavros_extras) for further info.
 
-## EKF2 Tuning for External Position Estimate
+## PX4 MAVLink Integration
 
-A set of parameters need to be set in order for EKF2 to use the external position estimation. You can use QGroundContorl (QGC) to easily adjust these parameters. All of the following parameters can be found in the EKF2 tab in QGC.
+PX4 uses the following MAVLink messages for getting external position information, and maps them to [uORB topics](http://dev.px4.io/en/middleware/uorb.html):
```

Comment:
@TSC21 Is there any advice on recommended streaming rates for these messages?

Reply:
Should be between 30 to 50 hz (30hz if containing covariances).

Reply:
Thanks. Captured in https://github.com/PX4/Devguide/pull/681/commits/b456f28a19b6ba16ca267bb28e2153d3ec9b48b4